python/mlc_llm/bench/request_processor.py

```diff
@@ -28,11 +29,22 @@ def __call__(self, request_records: List[RequestRecord]) -> List[RequestRecord]:
         return random.sample(request_records, self.num_requests)
 
 
-class AttachTimestamp(RequestProcessor):  # pylint: disable=too-few-public-methods
+class AttachRequestRateTimestamp(RequestProcessor):  # pylint: disable=too-few-public-methods
     """The processor that applies timestamps to the requests."""
 
+    def __init__(self, request_rate: Optional[np.float32]) -> None:
```

Comment:
make request_rate mandatory for this class. Instead, have a create pipeline that creates the request record processing pipeline, and optionally add AttachRequestRateTimestamp